src/subtyping.md

```diff
@@ -15,5 +15,71 @@ fn bar<'a>() {
     let t: &'a str = s;
 }
 ```
-Since `'static` "lives longer" than `'a`, `&'static str` is a subtype of
-`&'a str`.
+
+Since `'static` outlives `'a`, `&'static str` is a subtype of `&'a str`.
+
+Subtyping also exists for [higher-ranked types]. Replacing a higher ranked
+lifetime with a concrete lifetime produces a subtype.
```

Comment:
Nit: I don't believe the documentation you linked specifies what a "higher ranked type" is. Not entirely sure what you mean here. Literally just function pointers?

Reply:
Function pointers and trait objects